HTTPieCodeGenerator.coffee

```diff
@@ -111,8 +111,51 @@ HTTPieCodeGenerator = ->
                 s = ""
                 for key, value of object
                     sign = "#{if typeof(value) == 'string' then "=" else ":="}"
-                    s += "    #{addslashes key}#{sign}#{@json_body_object(value)} \\\n"
+                    if typeof value == 'object'
+                      s += addslashes(key) + sign
+                      if value.length != null
+                        s += @nested_array_object(value, 1)
+                    else
+                      s += '\'{\n' + @nested_hash_object(value, 1) + '}\''
         return s
+        
+    @nested_array_object = (array, level) ->
```

Comment:
Ah right, I see the trick of keeping all in one line. I don't know if it's worth the pain, there's a nice JSON serializer here for us, maybe it's better to just use it.